File: digital_wallet_laravel/app/Http/Controllers/AdminController.php
<?php

namespace App\Http\Controllers;

use App\Services\AdminService;
use Illuminate\Http\Request;

class AdminController extends Controller
{
    private AdminService $adminService;

    public function __construct(AdminService $adminService)
    {
        $this->adminService = $adminService;
    }

    public function listUsers(Request $request)
    {
        $search = $request->query('search', '');
        $page = (int) $request->query('page', 1);
        $size = (int) $request->query('size', 20);

        $result = $this->adminService->listUsers($search, $page, $size);

        return response()->json($result);
    }

    public function getUserDetail(Request $request, $id)
    {
        $result = $this->adminService->getUserDetail($id);

        return response()->json($result);
    }

    public function disableUser(Request $request, $id)
    {
        $this->adminService->disableUser($id);

        return response()->json([
            'status' => 'SUCCESS',
            'message' => 'User disabled successfully',
        ]);
    }

    public function enableUser(Request $request, $id)
    {
        $this->adminService->enableUser($id);

        return response()->json([
            'status' => 'SUCCESS',
            'message' => 'User enabled successfully',
        ]);
    }

    public function listTransactions(Request $request)
    {
        $page = (int) $request->query('page', 1);
        $size = (int) $request->query('size', 20);
        $username = $request->query('username', '');
        $startDate = $request->query('startDate', '');
        $endDate = $request->query('endDate', '');

        $result = $this->adminService->listTransactions($page, $size, $username, $startDate, $endDate);

        return response()->json($result);
    }

    public function getTransactionStats(Request $request)
    {
        $startDate = $request->query('startDate', '');
        $endDate = $request->query('endDate', '');

        $result = $this->adminService->getTransactionStats($startDate, $endDate);

        return response()->json($result);
    }
}

File: digital_wallet_springmvc/src/main/java/com/digitalwallet/controller/AdminController.java
package com.digitalwallet.controller;

import com.digitalwallet.model.*;
import com.digitalwallet.service.AdminService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/admin")
public class AdminController {

    private AdminService adminService;

    public void setAdminService(AdminService adminService) { this.adminService = adminService; }

    @GetMapping("/users")
    public ResponseEntity<PaginatedResponse<UserDTO>> listUsers(
            @RequestParam(defaultValue = "") String search,
            @RequestParam(defaultValue = "1") int page,
            @RequestParam(defaultValue = "20") int size) {
        return ResponseEntity.ok(adminService.listUsers(search, page, size));
    }

    @GetMapping("/users/{id}")
    public ResponseEntity<UserDetailDTO> getUserDetail(@PathVariable Long id) {
        return ResponseEntity.ok(adminService.getUserDetail(id));
    }

    @PutMapping("/users/{id}/disable")
    public ResponseEntity<ApiResponse> disableUser(@PathVariable Long id) {
        adminService.disableUser(id);
        return ResponseEntity.ok(ApiResponse.success("User disabled successfully"));
    }

    @PutMapping("/users/{id}/enable")
    public ResponseEntity<ApiResponse> enableUser(@PathVariable Long id) {
        adminService.enableUser(id);
        return ResponseEntity.ok(ApiResponse.success("User enabled successfully"));
    }

    @GetMapping("/transactions")
    public ResponseEntity<PaginatedResponse<AdminTransactionDTO>> listTransactions(
            @RequestParam(required = false) String username,
            @RequestParam(required = false) String from,
            @RequestParam(required = false) String to,
            @RequestParam(defaultValue = "1") int page,
            @RequestParam(defaultValue = "20") int size) {
        return ResponseEntity.ok(adminService.listTransactions(username, from, to, page, size));
    }

    @GetMapping("/transactions/stats")
    public ResponseEntity<TransactionStatsDTO> getTransactionStats(
            @RequestParam(required = false) String from,
            @RequestParam(required = false) String to) {
        return ResponseEntity.ok(adminService.getTransactionStats(from, to));
    }
}
